node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -1181,6 +1238,22 @@ where
 	Ok(())
 }
 
+/// Dequeue another collation fetch.
+async fn dequeue_next_collation_fetch(
```

Comment:
I understand why you named it that way.

However, I don't think that this fits. 

Maybe `dequeue_next_collation_and_fetch`?